content/docs/user-guide/checkpoints.md

```diff
@@ -0,0 +1,425 @@
+inside of the training epoch loop.
+
+```python
+for k, v in metrics.items():
+    print('Epoch %s: %s=%s'%(i, k, v))
+    dvclive.log(k, v)
+dvclive.next_step()
+```
+
+By adding the two `dvclive` methods to our training epoch loop, we are enabling
+checkpoints in our code and recording the training metrics at each of those
+checkpoints.
+
+Having the number of training epochs defined will prevent your training from
+running indefinitely. You can also hit `Ctrl + C` to stop creating new
+checkpoints if the number of epochs are not defined.
+
+### Manually adding checkpoints
+
+You also have the option to implement checkpoints in code with the
+`make_checkpoints()` method if you don't want to use `dvclive` in your code.
+This manually adds checkpoints for you, but it won't generate metrics for you.
+
+If you use `dvclive.next_step()`, you don't need to use `make_checkpoints()`.
+One or the other will give you checkpoints.
+
+_If you don't define a number of epochs to run, checkpoints will be created
+indefinitely until you terminate the code in the terminal. `Ctrl + C` will end
+the process._
+
+### Adding checkpoints conditionally
+
+You do have the option of adding checkpoints conditionally. You might be running
+thousands of training epochs and you only want to save after a number of epochs
+have run. Let's say you want to save every 100th checkpoint.
+
+Inside of the `main` method, where we have defined the training epochs, let's
+add this condition.
+
+```python
+# Evaluate and checkpoint.
+if i % 100 == 0:
+    metrics = evaluate(model, x_test, y_test)
+    for k, v in metrics.items():
+        dvclive.log(k, v)
+    dvclive.next_step()
+```
+
+Now we are conditionally saving checkpoints after a set number of runs have been
+completed. For the sake of this tutorial, we're going to remove that condition
+for everything going forward. We just wanted to make sure you know that you are
+able to do this and you don't have to save every checkpoint.
+
+## Running experiments
+
+With checkpoints enabled and working in our code, let's run the experiment
+again. You can run an experiment with the `dvc exp run` command. Since we've
+already run this code before, it will pick up training from the previous
+checkpoint and continue from there.
+
+You'll see output similar to this in your terminal while the training process is
+going on.
+
+```dvc
+Checkpoint experiment iteration '0ddae2a'.
+Updating lock file 'dvc.lock'
+Checkpoint experiment iteration '71d4d82'.
+Updating lock file 'dvc.lock'
+Checkpoint experiment iteration 'f59dca5'.
+Updating lock file 'dvc.lock'
+Checkpoint experiment iteration '08d7ab2'.
+Updating lock file 'dvc.lock'
+Checkpoint experiment iteration 'bcd3a62'.
+Updating lock file 'dvc.lock'
+Checkpoint experiment iteration '3cf9691'.
+Updating lock file 'dvc.lock'
+```
+
+Once your epochs are finished, it's time to take a look at the metrics we're
+working with.
```

Comment:
Terminating is a more common scenario. It might be better to ask users to terminate it after a few epochs (and increase\remove the epochs limit).